Content.Server/_DV/Grappling/EntitySystems/GrapplingSystem.cs

```diff
@@ -0,0 +1,504 @@
+    /// <param name="victim">Intended victim of the grapple.</param>
+    /// <returns>True if the grapple could start, false otherwise.</returns>
+    public bool CanGrapple(Entity<GrapplerComponent?> grappler, EntityUid victim)
+    {
+        if (!Resolve(grappler, ref grappler.Comp))
+            return false;
+
+        if (grappler.Comp.ActiveVictim.HasValue)
+            return false; // Can't grapple more than one target
+
+        if (_gameTiming.CurTime < grappler.Comp.CooldownEnd)
+            return false; // Cooldown on the grapple is not over yet
+
+        if (!TryComp<TagComponent>(victim, out var tagComp) ||
+            !_tagSystem.HasTag(tagComp, _grappleTargetId))
+            return false; // Not a valid target
+
+        if (!_actionBlockerSystem.CanInteract(grappler, victim))
+            return false;
+
+        return true;
+    }
+
+    /// <summary>
+    /// Returns whether a grapple is currently active on a victim.
+    /// </summary>
+    /// <param name="grappler">The grappler to check.</param>
+    /// <returns>True if there is an ongoing grapple, false otherwise.</returns>
+    public bool IsGrappling(Entity<GrapplerComponent?> grappler)
+    {
+        if (!Resolve(grappler, ref grappler.Comp))
+            return false; // Isn't a grappler, so nothing to do.
+
+        return grappler.Comp.ActiveVictim.HasValue;
+    }
+
+    /// <summary>
+    /// Attempts to start grappling a victim, rendering them unable to move and possibly
+    /// removing their ability to use their hands.
+    /// </summary>
+    /// <param name="grappler">Entity attempting to begin a grapple</param>
+    /// <param name="victim">Entity to be grappled.</param>
+    /// <returns>True if the grapple started, false otherwise.</returns>
+    public bool TryStartGrapple(Entity<GrapplerComponent?> grappler, EntityUid victim)
+    {
+        if (!Resolve(grappler, ref grappler.Comp))
+            return false;
+
+        if (!CanGrapple(grappler, victim))
+            return false;
+
+        if (!_interactionSystem.InRangeUnobstructed(grappler.Owner, victim))
+            return false;
+
+        StartGrapple((grappler, grappler.Comp), victim);
+        return true;
+    }
+
+    /// <summary>
+    /// Releases a victim from a grapple allowing them to move again and returning any
+    /// hands that were disabled.
+    /// </summary>
+    /// <param name="grappler">Entity, which was performing the grapple.</param>
+    /// <param name="manualRelease">Whether this release is performed by the grappler, or by the grappled.</param>
+    /// <returns>True if the grapple was released, false otherwise.</returns>
+    public bool ReleaseGrapple(Entity<GrapplerComponent?> grappler, bool manualRelease = false)
+    {
+        if (!Resolve(grappler, ref grappler.Comp))
+            return false;
+
+        var victim = grappler.Comp.ActiveVictim;
+        if (!victim.HasValue)
+            return false; // Not grappling anything
```

Comment:
is not {} victim

Reply:
```suggestion
        if (grappler.Comp.ActiveVictim is not { } victim)
            return false; // Not grappling anything
```
And then drop the `.Value` in the ReleaseGrapple-call below

Reply:
Literal dotnet magic